ui/hooks/useSignatureInsights.js

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { getSignatureOriginCaveat } from '@metamask/snaps-controllers';
+import { handleSnapRequest } from '../store/actions';
+import {
+  getSignatureInsightSnapIds,
+  getPermissionSubjectsDeepEqual,
+} from '../selectors';
+
+const SIGNATURE_INSIGHT_PERMISSION = 'endowment:signature-insight';
+
+export function useSignatureInsights({ txData }) {
+  const subjects = useSelector(getPermissionSubjectsDeepEqual);
+  const snapIds = useSelector(getSignatureInsightSnapIds);
+  const [loading, setLoading] = useState(true);
+  const [data, setData] = useState(undefined);
+
+  useEffect(() => {
+    let cancelled = false;
+
+    async function fetchInsight() {
+      setLoading(true);
+
+      const { msgParams } = txData;
+      const signature = {
+        from: msgParams.from,
+        data: JSON.parse(JSON.stringify(msgParams.data)),
```

Comment:
This seems flawed if `msgParams.data` is a string. I wonder if we should use the sanitized data object from the UI 🤔

Reply:
Updated